exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -100,7 +100,7 @@ public void setInitialCapacity(int numRecords) {
 
   @Override
   public int getValueCapacity() {
-    final int offsetValueCapacity = offsets.getValueCapacity() - 1;
+    final int offsetValueCapacity = Math.max(offsets.getValueCapacity() - 1, 0);
```

Comment:
Small comment, but there are two patterns used to handle special zero cases in the vector code. The one I have seen is actually for a slightly different case, the example copied below is out of the VarCharVector.setValueCount(int) method. It might be worth making the pattern consistent by using the ternary operator for both of these cases, so the fact that they are distinct is more obvious. This might seem a little contradictory, having them look totally different might be better. However, on first glance I thought this was a different way to express the same concept.

valueCount == 0 ? 0 : valueCount+1